packages/uicore/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -98,7 +100,7 @@ export function MultiSelect(props: MultiSelectProps): React.ReactElement {
         } else {
           onChange(selectedItems.concat(item))
         }
-        !avoidResetOnSelect && setQuery('')
+        ;(rest?.allowCreatingNewItems || !avoidResetOnSelect) && setQuery('')
```

Comment:
Can we change this to a standard `if` for readability?
```suggestion
        
        if (rest?.allowCreatingNewItems || !avoidResetOnSelect) {
          setQuery('')
        }
```